src/NodaTime/Globalization/NodaFormatInfo.cs

```diff
@@ -73,13 +78,30 @@ internal sealed class NodaFormatInfo
         private readonly Dictionary<Era, EraDescription> eraDescriptions;
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="NodaFormatInfo" /> class.
+        /// Initializes a new instance of the <see cref="NodaFormatInfo" /> class based solely
+        /// on a <see cref="System.Globalization.CultureInfo"/>.
         /// </summary>
-        /// <param name="cultureInfo">The culture info to base this on.</param>
+        /// <param name="cultureInfo">The culture info to use.</param>
+        [VisibleForTesting]
         internal NodaFormatInfo([NotNull] CultureInfo cultureInfo)
+            : this(cultureInfo, cultureInfo?.DateTimeFormat)
```

Comment:
This seems odd: are you deliberately allowing a `[NotNull]` parameter to be `null` here so that you can delegate to another constructor with `[NotNull]` parameters (which can verify null-ness) rather than fail here?